rosette/src/BaseSupp.cc

```diff
@@ -742,24 +744,29 @@ DEF("regexpCompare", regexpCompare, 2, 2) {
     CHECK(0, RBLstring, re);
     CHECK(1, RBLstring, str);
 
-    char *re_s, *str_s;
-    regexp* r;
     Ob* rv;
 
-    re_s = (char*)&re->byte(0);
-    str_s = (char*)&str->byte(0);
+    char* re_s = (char*)&re->byte(0);
+    char* str_s = (char*)&str->byte(0);
+
+    try {
+        std::regex r(re_s); // Compile the regexp
 
-    if ((r = GS_regcomp(re_s)) && GS_regexec(r, str_s)) {
-        int len = r->endp[0] - r->startp[0];
-        char* res = (char*)malloc(len + 1);
-        strncpy(res, r->startp[0], len);
-        res[len] = '\0';
-        rv = RBLstring::create(res);
-        free(res);
+        if (std::regex_match(str_s, r) == true) {   // See if it matches
+            const char *ret_str;
+            size_t len;
+            rv = RBLstring::create(str_s);
+        }
+        else {
+            rv = RBLFALSE;
+        }
+        return rv;
     }
-    else
+    catch(...) {
```

Comment:
This is why I hate exceptions. What could have been thrown? Why? Looking at the reference, I can't tell that any of these regex functions will throw an exception. There's no `throw` statements in any of the rosette code. Could this ever catch anything? If it did, shouldn't the program crash in preference to hiding some system exception behind an unlogged, untyped `catch` statement?

Reply:
http://en.cppreference.com/w/cpp/regex/regex_error

This page strongly implies that the std::regex constructor will throw exceptions on invalid expressions. It shows the method to figure out what was wrong with the expression, but in our case, the only thing we can do is catch the exception and return RBLFALSE. Do you want me to also spit out a warning with the error type as shown here?

Reply:
Can also return `PRIM_ERROR`.

Reply:
Done per our discussion